src/AWS/Orleans.Persistence.DynamoDB/Provider/DynamoDBGrainStorage.cs

```diff
@@ -61,7 +61,7 @@ public class DynamoDBGrainStorage : IGrainStorage, ILifecycleParticipant<ISiloLi
 
         public void Participate(ISiloLifecycle lifecycle)
         {
-            lifecycle.Subscribe(this.options.InitStage, Init, Close);
+            lifecycle.Subscribe(OptionFormattingUtilities.Name<DynamoDBGrainStorage>(this.name), this.options.InitStage, Init, Close);
```

Comment:
Should we move the call to `OptionFormattingUtilities.Name<DynamoDBGrainStorage>` in the `Subscribe` method?

Reply:
Not sure how that would work..?  Think it's probably better just to take a the string and rely on the components to build a user friendly name.